### DISCUSSION TITLE
Are match-filters in packageRules AND or OR conditions?

### DISCUSSION THREAD

**Discussion by ghdoergeloh** [renovatebot/renovate#10297] [Request Help] [answered]
I am new to renovate and I am a bit confused about how match-attributes in a packageRule work together.

As far as I understand, packageRules use match filter to set a behavior for specific updates while a update has different attributes:

- packageName
- updateType
- language
- datasource
- ...

So when I set a rule and add some filters those filters will check the attributes of that update.

What I am confused about is this: If I put multliple match-filters into one rule, will they work like a AND condition (which means, every filter has to match at the same time on the update) or is it a OR condition (which means, the update should at least fit one filter)?

For example I have seen something this:
 ```
 "packageRules": [
     {
       "matchPackagePatterns": ["angular"],
       "matchUpdateTypes": ["major"],
       "enabled": false
      }
  ]
```
which should prevent major updates of angular. So this means it must be a AND condition because the updates packagename has to contain "angular" and the updateType has to be "major".

But I could also imagine a case like this:

 ```
 "packageRules": [
     {
       "matchPackagePatterns": ["angular"],
       "matchPackageNames": ["zone"],
       "groupName": "angular update"
      }
  ]
```
The goal of this rule would be to join angular and zone updates into one "angular update" group. This means it is a OR condition because the packageName must either match "angular" OR be "zone". So is this a OR condition or do I have to change this packageRule to achive this behavior like either this solution:

 ```
 "packageRules": [
     {
       "matchPackagePatterns": ["angular", "^zone$"],
       "groupName": "angular update"
      }
  ]
```

or this:

 ```
 "packageRules": [
     {
       "matchPackagePatterns": ["angular"],
       "groupName": "angular update"
      },
     {
       "matchPackageNames": ["zone"],
       "groupName": "angular update"
      }
  ]
```

**Comment by rarkins** [comment, marked as answer]
`matchPackage*` fields are OR'd together while others are `AND`. The example of combining `matchPackagePatterns` and `matchPackageNames` works as you'd expect. In other words, this works:

```
"packageRules": [
    {
      "matchPackagePatterns": ["angular"],
      "matchPackageNames": ["zone"],
      "groupName": "angular update"
     }
 ]

```

**Reply by ghdoergeloh** [reply]
Ah, thx. I think I also found the passage in the code (  https://github.com/renovatebot/renovate/blob/9b458117f0fcc240f2a281cbc1527c7e09226515/lib/util/package-rules.ts#L121-L161 )

Is there anything in the documentation that I missed? Otherwise I would suggest to add this as a hint.

Another question (maybe a bit of topic but concerning the examples above): Is ist possible or even intended to use the groupName in different packageRules? So could I create a group "angular major update" by grouping all angular major updates with zone minor updates like this?:
```

"packageRules": [
    {
      "matchPackagePatterns": ["angular"],
      "matchUpdateTypes": ["major"],
      "groupName": "angular major update"
     },
    {
      "matchPackageNames": ["zone"],
      "matchUpdateTypes": ["major", "minor"],
      "groupName": "angular major update"
     }
 ]

```

**Reply by rarkins** [reply]
> Is there anything in the documentation that I missed? Otherwise I would suggest to add this as a hint.

In the docs we have this:

![image](https://user-images.githubusercontent.com/6311784/120627834-5b272d80-c464-11eb-88d6-9ee8e506ccfb.png)

> Another question (maybe a bit of topic but concerning the examples above): Is ist possible or even intended to use the groupName in different packageRules? So could I create a group "angular major update" by grouping all angular major updates with zone minor updates like this?:

Yes, you can. groupName does not need to be unique to a single package rule.

**Reply by ghdoergeloh** [reply]
Thanks for the quick replies!
